patches/server/0839-Don-t-tick-markers.patch

```diff
@@ -3,46 +3,47 @@ From: Noah van der Aa <ndvdaa@gmail.com>
 Date: Fri, 7 Jan 2022 11:58:26 +0100
 Subject: [PATCH] Don't tick markers
 
-Fixes https://github.com/PaperMC/Paper/issues/7276 by not adding markers to the entity
-tick list at all and ignoring them in Spigot's activation range checks. The entity tick
+Fixes https://github.com/PaperMC/Paper/issues/7276 and https://github.com/PaperMC/Paper/issues/8118
+by using a config option that, when set to false, does not add markers to the entity
+tick list at all and ignores them in Spigot's activation range checks. The entity tick
 list is only used in the tick and tickPassenger methods, so we can safely not add the
-markers to it.
+markers to it. When the config option is set to true, markers are ticked as normal.
 
 diff --git a/src/main/java/io/papermc/paper/command/subcommands/EntityCommand.java b/src/main/java/io/papermc/paper/command/subcommands/EntityCommand.java
-index ff99336e0b8131ae161cfa5c4fc83c6905e3dbc8..4ec1e8ded06c60d81446c67bba2aa8a04111fa9b 100644
+index ff99336e0b8131ae161cfa5c4fc83c6905e3dbc8..788b081fdab208769bb8ccc24dbe943a48504f32 100644
 --- a/src/main/java/io/papermc/paper/command/subcommands/EntityCommand.java
 +++ b/src/main/java/io/papermc/paper/command/subcommands/EntityCommand.java
 @@ -109,7 +109,7 @@ public final class EntityCommand implements PaperSubcommand {
                  ChunkPos chunk = e.chunkPosition();
                  info.left++;
                  info.right.put(chunk, info.right.getOrDefault(chunk, 0) + 1);
 -                if (!chunkProviderServer.isPositionTicking(e)) {
-+                if (!chunkProviderServer.isPositionTicking(e) || e instanceof net.minecraft.world.entity.Marker) { // Markers aren't ticked.
++                if (!chunkProviderServer.isPositionTicking(e) || (e instanceof net.minecraft.world.entity.Marker && !world.paperConfig().entities.markers.tick)) { // Markers aren't ticked by default.
                      nonEntityTicking.merge(key, 1, Integer::sum);
                  }
              });
 diff --git a/src/main/java/net/minecraft/server/level/ServerLevel.java b/src/main/java/net/minecraft/server/level/ServerLevel.java
-index 66b1ef69fe48340b5ccebd845b39f898515ff117..7ae780c4a38eb088f876e5bd03a0dba5836787fb 100644
+index 66b1ef69fe48340b5ccebd845b39f898515ff117..5051aea4f71c992d1884e4462435775af8beee4f 100644
 --- a/src/main/java/net/minecraft/server/level/ServerLevel.java
 +++ b/src/main/java/net/minecraft/server/level/ServerLevel.java
 @@ -2497,6 +2497,7 @@ public class ServerLevel extends Level implements WorldGenLevel {
          }
  
          public void onTickingStart(Entity entity) {
-+            if (entity instanceof net.minecraft.world.entity.Marker) return; // Paper - Don't tick markers
++            if (entity instanceof net.minecraft.world.entity.Marker && !paperConfig().entities.markers.tick) return; // Paper - Don't tick markers by default
```

Comment:
The paper commands should be changed to all be smth like `configurable marker ticking` instead of specify `Don't tick markers by default`. Since there is a configuration option, specifying what the default is just more annoying to change if we ever decide to change the default (like I suggested in the other comment).